File: app/src/main/java/io/treehouses/remote/pojo/ServiceInfo.java
package io.treehouses.remote.pojo;

public class ServiceInfo implements Comparable<ServiceInfo> {
    public final static int SERVICE_HEADER_INSTALLED = 0;
    public final static int SERVICE_RUNNING = 1;
    public final static int SERVICE_INSTALLED = 2;
    public final static int SERVICE_HEADER_AVAILABLE = 3;
    public final static int SERVICE_AVAILABLE = 4;

    public String name;
    public int serviceStatus;
    public String icon;
    public String info;
    public String autorun;

    //Headers
    public ServiceInfo(String n, int status) {
        this.name = n;
        this.serviceStatus = status;
    }

    //Services
    public ServiceInfo(String name, int serviceStatus, String icon, String info, String autorun) {
        this.name = name;
        this.serviceStatus = serviceStatus;
        this.icon = icon;
        this.info = info;
        this.autorun = autorun;
    }

    @Override
    public int compareTo(ServiceInfo o) {
        return this.serviceStatus - o.serviceStatus;
    }

    public boolean isHeader() {
        return serviceStatus == SERVICE_HEADER_AVAILABLE || serviceStatus == SERVICE_HEADER_INSTALLED;
    }
}

File: app/src/main/java/io/treehouses/remote/pojo/ServiceInfo.kt
package io.treehouses.remote.pojo

import java.io.Serializable

class ServiceInfo : Comparable<ServiceInfo>, Serializable {
    @JvmField
    var name: String
    @JvmField
    var serviceStatus: Int
    @JvmField
    var icon: String? = null
    @JvmField
    var info: String? = null
    @JvmField
    var autorun: String? = null

    //Headers
    constructor(n: String, status: Int) {
        name = n
        serviceStatus = status
    }

    //Services
    constructor(name: String, serviceStatus: Int, icon: String?, info: String?, autorun: String?) {
        this.name = name
        this.serviceStatus = serviceStatus
        this.icon = icon
        this.info = info
        this.autorun = autorun
    }

    override fun compareTo(other: ServiceInfo): Int {
        return serviceStatus - other.serviceStatus
    }

    val isHeader: Boolean
        get() = serviceStatus == SERVICE_HEADER_AVAILABLE || serviceStatus == SERVICE_HEADER_INSTALLED

    companion object {
        const val SERVICE_HEADER_INSTALLED = 0
        const val SERVICE_RUNNING = 1
        const val SERVICE_INSTALLED = 2
        const val SERVICE_HEADER_AVAILABLE = 3
        const val SERVICE_AVAILABLE = 4
    }
}